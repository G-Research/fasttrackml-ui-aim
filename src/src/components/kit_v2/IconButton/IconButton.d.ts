import React from 'react';

import { CSS, ColorPaletteType } from 'config/stitches/types';

import { ButtonVariantType } from '../Button';
export interface IIconButtonProps
  extends Partial<React.HTMLAttributes<HTMLButtonElement>> {
  /**
   * @description The name of the icon
   * @example 'add'
   */
  icon: React.ReactNode;
  /**
   * @description The color of the icon button
   * @example 'primary'
   */
  color?: ColorPaletteType;
  /**
   * @description The size of the icon button
   * @example 'md'
   * @default 'md'
   */
  size?: sizeType;
  /**
   * @description The variant of the icon button
   * @example 'contained'
   * @default 'contained'
   */
  variant?: ButtonVariantType;
  /**
   * @description The disabled state of the icon button
   * @example false
   * @default false
   */
  disabled?: boolean;
  /**
   * @description The css prop of the icon button
   * @example { backgroundColor: 'red' }
   * @default {}
   */
  css?: CSS;
}

// IconButton component size types
type sizeType = 'xs' | 'sm' | 'md' | 'lg' | 'xl';

// IconButton component variants
