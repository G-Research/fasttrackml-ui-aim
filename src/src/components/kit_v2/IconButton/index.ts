import IconButton from './IconButton';

export * from './IconButton.d';
export default IconButton;
