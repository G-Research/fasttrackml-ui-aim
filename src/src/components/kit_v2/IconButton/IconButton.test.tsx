import { render } from '@testing-library/react';

import IconButton from './IconButton';

// Test of IconButton component
describe('<IconButton icon="search" />', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<IconButton icon='search' />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test onClick event
  test('onClick event works properly', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <IconButton icon='search' onClick={onClick} />,
    );
    const button = getByTestId('icon-button');
    button.click();
    expect(onClick).toHaveBeenCalled();
  });
});
