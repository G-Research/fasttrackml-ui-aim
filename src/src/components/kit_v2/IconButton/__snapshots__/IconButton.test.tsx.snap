// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton icon="search" /> Renders Correctly 1`] = `
<DocumentFragment>
  <button
    class="AimUi-c-sFyLx AimUi-c-sFyLx-fDXBFo-size-sm AimUi-c-sFyLx-ihtltCK-css"
    data-testid="icon-button"
  >
    <i
      class="Icon__container icon-search "
    />
  </button>
</DocumentFragment>
`;
